src/wovnio/utils/request_handlers/CurlRequestHandler.php

```diff
@@ -50,7 +50,13 @@ protected function post($url, $request_headers, $data, $timeout)
         $response = curl_exec($curl_session);
         $header_size = curl_getinfo($curl_session, CURLINFO_HEADER_SIZE);
         $headers = $response ? explode("\r\n", substr($response, 0, $header_size)) : array();
-
+        $parsedHeaders = array('status' => $headers[0]);
+        foreach ($headers as $value) {
+            $exploded = explode(':', $value, 2);
+            if (sizeof($exploded) == 2) {
+                $parsedHeaders[trim($exploded[0])] = trim($exploded[1]);
+            }
+        }
         if (curl_error($curl_session) !== '') {
```

Comment:
How about making `function parseHeader(response, header_size)`.

Reply:
That's a good idea. I stole this part from the test utils. I guess it's time we make a common util to put this kind of common code in.